app/assets/javascripts/va_facilities/facilities_utilities.es6

```diff
@@ -0,0 +1,139 @@
+    }
+  })
+}
+
+function attachVisnSelectListener() {
+  $(visnSelect).on('change', () => {
+    _setReloadData(true);
+    let visn = parseInt($(visnSelect).val());
+    let complexity = $(complexitySelect).val();
+    isFacilityFilter = false;
+    if (visn || complexity) {
+      _sendAjaxRequest({ visn, complexity });
+    }
+  });
+}
+
+function attachComplexitySelectListener() {
+  $(complexitySelect).on('change', () => {
+    _setReloadData(true);
+    let visn = parseInt($(visnSelect).val());
+    let complexity = $(complexitySelect).val();
+    isFacilityFilter = false;
+    if (visn || complexity) {
+      _sendAjaxRequest({ visn, complexity });
+    }
+  })
+}
+
+function _sendAjaxRequest(data = null) {
+  _toggleSpinners({ displaySpinner: true })
+  if (_getReloadData()) {
+    $.ajax({
+      type: "GET",
+      data: data,
+      dataType: "json",
+      url: "/facilities/load-facilities-rows",
+      success: function (data) {
+        $(tableRows).empty();
+        if (data.count > 0) {
+          $(table).removeClass("display-none");
+          $(tableRows).append(data.rowsHtml);
+        } else {
+          $(noResults).removeClass("display-none");
+        }
+        // if facility combo box is selected
+        if (isFacilityFilter) {
+          _clearSelect();
+        } else {
+          autoClearClicked = true;
+          $(comboBoxClearBtn).click();
+        }
+        // reset to defaults
+        autoClearClicked = false;
+        isFacilityFilter = true;
+        _setCounterText(data.count);
+        _toggleSpinners({ displaySpinner: false });
+        _setReloadData(false);
+      },
+    });
+  }
+}
+
+function attachComboBoxClearArrivalListener() {
+  $(document).arrive(comboBoxClearBtn, (input) => {
+    $(input).on("click", () => {
+      _setReloadData(true);
+      // makes sure we don't send another request when we automatically cleared the combo box input
+      if (!autoClearClicked) {
+        _sendAjaxRequest();
+      }
+    });
+  });
+}
+
+function _clearSelect() {
+  $(visnSelect).val("");
+  $(complexitySelect).val("");
+}
+
+function _setCounterText(count) {
+  let counterText = `Displaying ${count} result${count === 1 ? '' : 's'}:`;
+  $('.dm-results-count').text(counterText);
+}
+
+function _toggleSpinners({ displaySpinner }) {
+  if (displaySpinner) {
+    $(loadingSpinner).removeClass("display-none");
+    $(facilitiesIndexView).addClass("display-none");
+    $(noResults).addClass("display-none");
+    $(table).addClass("display-none");
+  } else {
+    $(loadingSpinner).addClass("display-none");
+    $(facilitiesIndexView).removeClass("display-none");
+    $(table).removeClass("display-none");
+  }
+}
+
+function _setReloadData(bool) {
+  $(facilitiesIndex).data("reload", bool);
+}
+
+function _getReloadData() {
+  return $(facilitiesIndex).data("reload");
+}
+
+function execFacilitiesIndexFns() {
+  // make sure we don't reload the page when we click links on the facilities index page
```

Comment:
⚠️ for some reason, when clicking links like a link to the facility in the table it would reload the page before redirecting to the facility show page. In order to address this, the check was added.